README.md

```diff
@@ -1 +1,523 @@
+│      IPVM Config, Verification Level, etc     │                           │
+│                 (IPVM Task)                   │                           │
+│                                               │                           │
+├───────────────────────────────────────────────┴───────────────────────────┤
+│                                                                           │
+│                                 Call Graph                                │
+│                             (UCAN Invocation)                             │
+│                                                                           │
+├───────────────────────────────────────────────────────────────────────────┤
+│                                                                           │
+│                                 Authority                                 │
+│                                (UCAN Core)                                │
+│                                                                           │
+└───────────────────────────────────────────────────────────────────────────┘
+```
+
+# 2 Envelope
+
+The outer wrapper of a workflow MUST contain the following fields:
+
+| Field           | Type       | Description                                                             | Required |
+|-----------------|------------|-------------------------------------------------------------------------|----------|
+| `ipvm/workflow` | `Workflow` | IPVM Workflow                                                           | Yes      |
+| `signature`     | `VarSig`   | [VarSig](https://github.com/ChainAgnostic/varsig/) of serialized fields | Yes      |
+
+| Field      | Type                         | Description                                                            | Required | Default |
+|------------|------------------------------|------------------------------------------------------------------------|----------|---------|
+| `v`        | `"0.1.0"`                    | IPVM workflow version                                                  | Yes      |         |
+| `meta`     | `{String : Any}`             | User-defined object (tags, comments, etc)                              | No       | `{}`    |
+| `parent`   | `[&Workflow, Label] or Null` | The workflow & task label that initiated the current workflow (if any) | No       | `Null`  |
+| `config`   | `Config`                     | Global configuration (e.g. timeout for the entire workflow)            | No       | `{}`    |
+| `defaults` | `Config`                     | Individual task config defaults                                        | No       | `{}`    |
+| `tasks`    | `UCAN.Invocation`            | UCAN Invocation                                                        | Yes      |         |
+| `catch`    | `&WasmTask`                  | Deterministic Wasm that fires on exceptions                            | No       | `{}`    |
```

Comment:
so is `wasmtask` a ucan/invocation task, as it defaults to a `{}`? I ask because elsewhere in the doc, it looks like a CID directly. I wonder if `catch` points to a CID or wasm::CID and/or to a known top-level key/name of an invocation task that may already have been introduced previously?

Reply:
So, here `catch` is wasm + determinism-specific, so it should not be allowed on the workflow top-level if an inner task is a docker one.

Reply:
> as it defaults to a {}

Oops! That should definitely default to `Null`. Will fix!

> a known top-level key/name of an invocation task 

I want to keep it in a separate block for a couple reasons:

* Pure Wasm only
* Set really tight, enforced resource restrictions (it should terminate quickly)
* If it's a named task, it will get scheduled with the rest of the workflow

Reply:
I need to tighten up some of these, too. That really should be `&Wasm`.